src/parser/jobs/rpr/modules/ArcaneCircle.tsx

```diff
@@ -18,6 +20,13 @@ export class ArcaneCircle extends BuffWindow {
 	override initialise() {
 		super.initialise()
 
+		// Multiple reapers can override each other's buffs
+		const dupeFilter = filter<Event>().source(noneOf([this.parser.actor.id]))
+			.target(this.parser.actor.id)
+			.status(this.buffStatus.id)
+
+		this.addEventHook(dupeFilter.type('statusApply'), this.maybeReOpenPreviousWindow)
```

Comment:
Moved the logic to BuffWindow and put it behind an option that jobs can set for themselves. I'm not sure what you mean by an option for window collapsing, but I think it's more readable to have two hooks (for player-owned IDs, and not) and do flow control there, rather than a single hook that then has conditionals to handle whodunit.